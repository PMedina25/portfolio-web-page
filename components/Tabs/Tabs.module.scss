.tabs {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding-bottom: rem(40);

  @include breakpoint(desktop) {
    flex-direction: row-reverse;
    align-items: center;
    gap: rem(80);
    text-align: justify;
  }
}
