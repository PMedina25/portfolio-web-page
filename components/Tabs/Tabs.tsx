import React, { useState } from "react";

import TabNav from "./TabNav/TabNav";
import TabContent from "./TabContent/TabContent";

export interface Tab {
  label: string;
  content: React.ReactElement;
}

interface TabsProps {
  tabs: Tab[];
}

const Tabs = ({ tabs }: TabsProps) => {
  const [activeTab, setActiveTab] = useState(0);

  const onTabClick = (index: number) => {
    setActiveTab(index);
  };

  return (
    <div>
      <TabNav activeTab={activeTab} onTabClick={onTabClick} tabs={tabs} />
      <TabContent activeTab={activeTab} tabs={tabs} />
    </div>
  );
};

export default Tabs;
