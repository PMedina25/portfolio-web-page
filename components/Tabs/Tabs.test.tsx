import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import Tabs from "./Tabs";

describe("Tabs component", () => {
  const tabs = [
    {
      label: "Tab 1",
      content: <div>Content for Tab 1</div>,
    },
    {
      label: "Tab 2",
      content: <div>Content for Tab 2</div>,
    },
  ];

  it("renders correctly (match snapshot)", () => {
    const tree = renderer.create(<Tabs tabs={tabs} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the first tab by default", () => {
    const { getByText } = render(<Tabs tabs={tabs} />);
    expect(getByText("Content for Tab 1")).toBeInTheDocument();
  });

  it("renders the selected tab", () => {
    const { getByText, getByRole } = render(<Tabs tabs={tabs} />);
    const tabButton = getByRole("button", { name: "Tab 2" });
    fireEvent.click(tabButton);
    expect(getByText("Content for Tab 2")).toBeInTheDocument();
  });
});
