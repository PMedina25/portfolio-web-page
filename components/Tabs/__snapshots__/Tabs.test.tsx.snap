// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders correctly (match snapshot) 1`] = `
<div>
  <div
    className="tab-nav"
  >
    <button
      className="active"
      onClick={[Function]}
    >
      Tab 1
    </button>
    <button
      className=""
      onClick={[Function]}
    >
      Tab 2
    </button>
  </div>
  <div
    className="tab-content"
  >
    <div>
      Content for Tab 1
    </div>
  </div>
</div>
`;
